spec/models/job_run_spec.rb

```diff
@@ -25,15 +26,33 @@
       expect(JobRun.new).not_to be_valid
       expect(discovery_report_run).to be_valid
     end
+
     context "#job_type enum" do
       it "defines expected values" do
         is_expected.to define_enum_for(:job_type).with(
           "discovery_report" => 0,
-          "pre_assembly" => 1
+          "preassembly" => 1
         )
       end
     end
 
     it { is_expected.to belong_to(:bundle_context) }
+
+    context 'job_type' do
+      it 'valid values' do
+        expect(JobRun.new(job_type: 0, bundle_context: bc)).to be_valid
+        expect(JobRun.new(job_type: 1, bundle_context: bc)).to be_valid
+        expect(JobRun.new(job_type: 'discovery_report', bundle_context: bc)).to be_valid
+        expect(JobRun.new(job_type: 'preassembly', bundle_context: bc)).to be_valid
```

Comment:
I think these are duplicative of the above test that identifies the same keys and values as defined in the `job_type` enum.  Values defined in the enum must be valid values or rails is broken, not our app, right?

Reply:
Since these tests do not block merging, I'm going ahead and approve this PR

Reply:
I looked around to see whether a hash enum's keys or its values were valid;  didn't find anything quickly and ended up writing the test to figure out empirically it was both keys and values.  I also didn't know declaring an enum was sufficient for validation purposes.  Perhaps everyone but me already knew, perhaps not.  Perhaps all future maintainers of the code will know this;  perhaps not.

Reply:
I recall the team had a similar conversation on the last project when we first used enum.  My understanding is that the rest of the team is pretty confident using enums now.  I think it is commonplace rails and not deeply "magical".  

My point here is that the latter tests are testing "how do enums work?" and not "how is this app defined?"  This mirrors the conversations that @jcoyne and I had on HyBox work, except I was the one acquiring knowledge and advocating tests similar to yours and he (successfully) argued against testing rails internals.

As far as future maintainers, if they are completely unfamiliar with rails, it isn't the scope of this application or its tests to educate them (or us).  We are certain to do a partial, dated, and poor job of it if we accepted that scope!